src/storage/fs/index.ts

```diff
@@ -0,0 +1,15 @@
+import { isNodeJS } from '../../utils/environment'
+import { Constructor } from '../../utils/mixins'
+import { FileSystem } from './FileSystem'
+
+let FileSystemClass: Constructor<FileSystem>
+
+if (isNodeJS()) {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  FileSystemClass = require('./NodeFileSystem').NodeFileSystem
```

Comment:
Not sure if this is gonna work in RN. I think that the RN metro bundler tries to load everything which is required somewhere in the code. This condition is then called only in runtime

Reply:
Hmm didn't think about it. Seems complex: https://reactnative.dev/docs/ram-bundles-inline-requires#inline-requires

If we already want to move to multiple packages the best way for now I think would be to export both classes and require it to be passed as a config:

We could do the same for the `fetch` method so we don't need to depend on `node-fetch`.

```ts
new Agent({
	fileSystem: new ReactNativeFileSystem()
})
```